projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -402,15 +402,18 @@ public function add_jetpack_menu() {
 	 * Update Site Editor menu item's link and position.
 	 */
 	public function add_gutenberg_menus() {
-		if ( self::CLASSIC_VIEW === $this->get_preferred_view( 'admin.php?page=gutenberg-edit-site' ) ) {
+		if ( self::CLASSIC_VIEW === $this->get_preferred_view( 'site-editor.php' ) ) {
```

Comment:
Do you think we should try to get the preference from the old route if the one for `site-editor-php` is `null` _(I don't know what is the actual return value returned when it can't find a persisted preferred view, though, would have to look)_?

I think that if we don't, users will lose their preference here, if it was set 🤔

Reply:
That makes sense to me 👍

Reply:
Do you know how to set this preference for the site editor? 🤔 cc @Addison-Stavlo

Reply:
I did a little searching trying to understand this. The "preferred view" comes from the `preferred-view` query parameter in the url. The only place I can see that we set that in the UI is in the View panel at the top right of some Calypso/wp-admin screens.

<img width="541" alt="image" src="https://user-images.githubusercontent.com/1699996/181847102-a26638b2-1070-4fe7-a41f-c45e42ccdc1a.png">

https://github.com/Automattic/jetpack/blob/6e9b58d2e829c436a67db7050b9b5915cd1fb730/projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php#L553

I'm not seeing a way the preferred view would get set specifically for the Site editor, other than manually appending the query parameter to the wp-admin url. If that's correct, I think we need to worry about persisting the preference.

Reply:
I miss-typed that last comment! Edited, it now reads: "I **don't** think we need to worry about persisting the preference."

Reply:
Hey @creativecoder, thanks for diving into this! Yeah, I was having a chat with @jeyip about the very same thing and we reached the same conclusion. It seems that we can assume the Site Editor has always been set to load inside the Gutenframe and that users couldn't change that.

Reply:
That being said, I'll leave an additional note that - as reported by you [here](https://github.com/Automattic/wp-calypso/issues/66020) - the Gutenframe isn't able to load the Site Editor in AT sites at the moment, though.